[mono] Fix output directory for cross compilers if OutputRID is overridden

akoeplinger opened the pull request:
This happens e.g. in the VMR where OutputRID is set globally. We assumed in monoaotcross.proj that the inner mono cross compiler runtime build would put files into a path that matches the target RID but we were using OutputRID in the inner build.

Instead use the preexisting MonoAotCrossDir property.

Also fix marking DotNetBuildMonoCrossAOT=true as a ShortStack build.

ViktorHofer commented on src/mono/monoaotcross.proj:
```diff
@@ -43,20 +43,23 @@
       <MonoAotTargetArchitecture>$(MonoAotTargetRid.Substring($([MSBuild]::Add(1, $(MonoAotTargetRid.LastIndexOf('-'))))))</MonoAotTargetArchitecture>
       <MonoAotHostOS>$(TargetOS)</MonoAotHostOS>
       <MonoAotHostOS Condition="'$(TargetsLinuxMusl)' == 'true'">linux-musl</MonoAotHostOS>
+      <MonoAotHostArchitecture>$(TargetArchitecture)</MonoAotHostArchitecture>
```
Doesn't this override L43 unconditionally? Is that the intent?

akoeplinger replied:
this is different, host vs target

ViktorHofer replied:
Ok got it. Shouldn't the host architecture default to the target architecture? That's at least what we do in the VMR: https://github.com/dotnet/sdk/blob/dcaddf212a5254439ba4fac8b5efc1bf56c2b52b/src/SourceBuild/content/Directory.Build.props#L48-L49

ViktorHofer replied:
Oh I see. These properties are being used differently here. Interesting.

akoeplinger committed:
[mono] Fix output directory for cross compilers if OutputRID is overridden

This happens e.g. in the VMR where OutputRID is set globally. We assumed in monoaotcross.proj that the inner mono cross compiler runtime build would put files into a path that matches the target RID but we were using OutputRID in the inner build.

Instead use the preexisting MonoAotCrossDir property.

Also fix marking DotNetBuildMonoCrossAOT=true as a ShortStack build.

akoeplinger commented:
/ba-g timeouts are unrelated
